src/map/script.c

```diff
@@ -8623,6 +8628,43 @@ BUILDIN(readparam) {
 	return true;
 }
 
+BUILDIN(setparam)
+{
+	int type;
+	struct map_session_data *sd;
+	struct script_data *data = script_getdata(st, 2);
+	int val = script_getnum(st, 3);
+
+	if (data_isreference(data) && reference_toparam(data)) {
+		type = reference_getparamtype(data);
+	} else {
+		type = script->conv_num(st, data);
+	}
```

Comment:
nonono, what I meant was ....
ok I try do this myself ...

```
	if (data_isreference(data) && reference_toparam(data)) {
		type = reference_getparamtype(data);
	} else if (data->type == C_INT) {
		type = script->conv_num(st, data);
	} else {
		ShowWarning("buildin_setparam: Not a parameter.\n");
		script->reportdata(data);
		return false;
	}
```
tested with script
```
prontera,155,187,5	script	kjsdhfksjdhf	1_F_MARIA,{
	setparam asdf, 100;
	end;
}
```
print error as
```
[Warning]: buildin_setparam: Not a parameter.
[Debug]: Data: variable name='asdf' index=0
[Debug]: Source (NPC): kjsdhfksjdhf at prontera (155,187)
```

Reply:
sry this one no need, perhaps sometimes people want to do a loop with this
ooops !